fe/fe-core/src/main/java/com/starrocks/alter/Alter.java

```diff
@@ -188,6 +198,135 @@ public void processDropMaterializedView(DropMaterializedViewStmt stmt) throws Dd
+                        + "Do not allow to do ALTER ops");
+            }
+
+            //rename materialized view
+            if (newMvName != null) {
+                processRenameMaterializedView(oldMvName, newMvName, db, materializedView);
+            } else if (refreshSchemeDesc != null) {
+                //change refresh scheme
+                processChangeRefreshScheme(refreshSchemeDesc, materializedView);
+            } else {
+                throw new DdlException("unsupported modification for materialized view");
+            }
+        } finally {
+            db.writeUnlock();
+        }
+    }
+
+    private void processChangeRefreshScheme(RefreshSchemeDesc refreshSchemeDesc, MaterializedView materializedView)
+            throws DdlException {
+        final MaterializedView.MvRefreshScheme refreshScheme = materializedView.getRefreshScheme();
+        final String refreshType = refreshSchemeDesc.getType().name();
+
+        if (refreshSchemeDesc instanceof SyncRefreshSchemeDesc) {
+            throw new DdlException("unsupported change to SYNC refresh type");
+        }
+
+        if (refreshSchemeDesc instanceof AsyncRefreshSchemeDesc) {
+            AsyncRefreshSchemeDesc asyncRefreshSchemeDesc = (AsyncRefreshSchemeDesc) refreshSchemeDesc;
+            final MaterializedView.AsyncRefreshContext asyncRefreshContext = refreshScheme.getAsyncRefreshContext();
+            asyncRefreshContext.setStartTime(Utils.getLongFromDateTime(asyncRefreshSchemeDesc.getStartTime()));
+            asyncRefreshContext.setStep(asyncRefreshSchemeDesc.getStep());
+            asyncRefreshContext.setTimeUnit(asyncRefreshSchemeDesc.getTimeUnit());
+        }
+
+        final RefreshType newRefreshType = RefreshType.valueOf(refreshType);
+        refreshScheme.setType(newRefreshType);
+        final ChangeMaterializedViewRefreshSchemeLog log = new ChangeMaterializedViewRefreshSchemeLog(materializedView);
+        EditLog editLog = GlobalStateMgr.getCurrentState().getEditLog();
+        editLog.logMvChangeRefreshScheme(log);
+    }
+
+    private void processRenameMaterializedView(String oldMvName, String newMvName, Database db,
+                                               MaterializedView materializedView) throws DdlException {
+        if (oldMvName.equals(newMvName)) {
+            throw new DdlException("Same materialized view name");
+        }
+        if (db.getTable(newMvName) != null) {
+            throw new DdlException("Materialized view [" + newMvName + "] is already used");
+        }
+        materializedView.setName(newMvName);
+        db.dropTable(oldMvName);
+        db.createTable(materializedView);
+        final RenameMaterializedViewLog renameMaterializedViewLog =
+                new RenameMaterializedViewLog(materializedView, oldMvName, newMvName);
+        EditLog editLog = GlobalStateMgr.getCurrentState().getEditLog();
```

Comment:
I think we could put the EditLog.logxxx method to the outside of DB lock. because the log operation maybe need a long time.